test/functional/feature_assumeutxo.py

```diff
@@ -204,6 +200,35 @@ def test_snapshot_with_less_work(self, dump_output_path):
             assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", node.loadtxoutset, dump_output_path)
         self.restart_node(0, extra_args=self.extra_args[0])
 
+    def test_snapshot_in_a_divergent_chain(self, dump_output_path):
+        # First rollback node2's chain to the pregenerated one, up to height 199
+        node = self.nodes[2]
+        block_hash = node.getblockhash(START_HEIGHT + 1)
+        node.invalidateblock(block_hash)
+        assert_equal(node.getblockcount(), START_HEIGHT)
+
+        self.log.info(f"Check importing a snapshot where current chain-tip is not an ancestor of the snapshot block but has less work")
+        # Generate a divergent chain in node2 but with less work compared to the snapshot
+        self.generate(node, nblocks=99, sync_fun=self.no_op)
+        assert node.getblockcount() < SNAPSHOT_BASE_HEIGHT
+        # Try importing the snapshot and assert its success
+        loaded = node.loadtxoutset(dump_output_path)
+        assert_equal(loaded['coins_loaded'], SNAPSHOT_BASE_HEIGHT)
+        assert_equal(loaded['base_height'], SNAPSHOT_BASE_HEIGHT)
+
+        # Restart the node and delete the snapshot chainstate from previous test.
+        self.restart_node(2, extra_args=['-reindex-chainstate=1', *self.extra_args[2]])
+        assert node.getblockcount() < SNAPSHOT_BASE_HEIGHT
+
+        self.log.info(f"Check importing a snapshot where current chain-tip is not an ancestor or a descendant of the snapshot block and has more work")
+        # Generate two extra blocks and make sure node2's chain has more work than the snapshot's chain
+        # This covers the scenario where the snapshot block is not on the most-work chain
+        self.generate(node, nblocks=2, sync_fun=self.no_op)
+        assert node.getblockcount() > SNAPSHOT_BASE_HEIGHT
```

Comment:
I would prefer usage of `assert_equal` with the precise height you expect here.

Reply:
Fixed